code/game/atoms.dm

```diff
@@ -280,22 +280,22 @@ var/list/blood_splatter_icons = list()
 
 //returns 1 if made bloody, returns 0 otherwise
 /atom/proc/add_blood(mob/living/carbon/M)
+	if(!M || !M.has_dna() || rejects_blood())
+		return 0

```

Comment:
The runtime isn't linked to dna, it only happened because I removed some checks that shouldn't have been necessary. The proc runtimes because M is sometimes null.